src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,175 @@
+        let replacedValueLines = [];
+        for (let defaultLine of newValueLines) {
+            // Skipped lines
+            if (defaultLineSet.has(defaultLine.trimStart())) {
+                continue;
+            } else if (defaultLine.trimStart() === defaultBindingResult) { // Result change
+                replacedValueLines.push(defaultLine.replace(defaultBindingResult, sqlBindingResult));
+            } else {
+                // Normal lines to be included
+                replacedValueLines.push(defaultLine);
+            }
+        }
+        newValue = replacedValueLines.join(os.EOL);
+        fs.writeFileSync(filePath, newValue, 'utf-8');
+    }
+
+    private async isAzureFunctionProjectOpen(): Promise<boolean> {
+        if (vscode.workspace.workspaceFolders === undefined || vscode.workspace.workspaceFolders.length === 0) { return false; }
+        const projFile = await this.getProjectFile();
+        const hostFile = await this.getHostFile();
+        return projFile !== undefined && hostFile !== undefined;
+    }
+
+    private async getProjectFile(): Promise<string | undefined> {
+        const projFiles = await vscode.workspace.findFiles('**/*.csproj');
+        return projFiles.length > 0 ? projFiles[0].fsPath : undefined;
+    }
+
+    private async getHostFile(): Promise<string | undefined> {
+        const hostFiles = await vscode.workspace.findFiles('**/host.json');
+        return hostFiles.length > 0 ? hostFiles[0].fsPath : undefined;
+    }
+
+    private async getSettingsFile(): Promise<string | undefined> {
+        const settingsFiles = await vscode.workspace.findFiles('**/local.settings.json');
+        return settingsFiles.length > 0 ? settingsFiles[0].fsPath : undefined;
+    }
+
+    private getNewFunctionFile(): Promise<string> {
+        return new Promise((resolve) => {
+            const watcher = vscode.workspace.createFileSystemWatcher(new vscode.RelativePattern(vscode.workspace.workspaceFolders[0], '**/*.cs'), false, true, true);
+            watcher.onDidCreate((e) => {
+                resolve(e.fsPath);
+                watcher.dispose();
+            });
+        });
+    }
+
+    private async addNugetReferenceToProjectFile(): Promise<void> {
+        // Make sure the nuget source is added
+        const currentSources = await this.executeCommand('dotnet nuget list source');
+        if (currentSources.indexOf(sqlBindingNugetSource) === -1) {
+            await this.executeCommand(`dotnet nuget add source ${sqlBindingNugetSource}`);
+        }
+        const projFile = await this.getProjectFile();
+        await this.executeCommand(`dotnet add package ${sqlBindingPackageName} --version ${sqlBindingPackageVersion}`, path.dirname(projFile));
+    }
+
+    private async addConnectionStringToConfig(connectionString: string): Promise<void> {
+        const settingsFile = await this.getSettingsFile();
+        const content = await fs.promises.readFile(settingsFile);
```

Comment:
Does there need to be a check if settingsFile is undefined since `getSettingsFile()` can return undefined?